src/string.cr

```diff
@@ -509,7 +509,7 @@ class String
     end
   end
 
-  private def to_u64_info(base, whitespace, underscore, prefix, strict)
+  private macro to_u64_info(base, whitespace, underscore, prefix, strict, unsigned)
```

Comment:
why does this need to be a macro instead of a method?

Reply:
See https://github.com/crystal-lang/crystal/pull/7446/files#diff-87ebb896824e7dc3bd8d5b29293bcf00R529

Reply:
Does it have to be a maco. it seems like it could just check for the parameter.

Reply:
FWIW it's about trading runtime for compile-time check.

Reply:
This is a no-no, it duplicates the binary size and compile time for all methods.

Reply:
And probably optimized just the same by LLVM anyway.

Reply:
> And probably optimized just the same by LLVM anyway

The new `to_i` methods were always something like `1.04× slower` on negative numbers when I ran the benchmark about 6 times in a row. But it could be a coincidence.

> This is a no-no, it duplicates the binary size and compile time for all methods.

But how does it duplicate the binary size? Isn't it even less binary size with a macro? A macro is always evaluated/inlined by Crystal but for a method the chance that it gets inlined isn't always 100% as far as I know.
But I understand the compile-time part and will revert this. The compile times take too long already.

Reply:
Expanded macros and inlined methods multiply the amount of code because it's repeated for each time they're used. On the contrary, the code of non-inlined methods is shared between all it's callers.